pyproject.toml

```diff
@@ -1,48 +1,179 @@
 [build-system]
 requires = [
-    "setuptools>=42.0",
-    "setuptools_scm[toml]>=3.4"
+    "hatchling>=1.4.1",
```

Comment:
Why 1.4.1? I think 1.0 dropped 3.6 when 2.7 was dropped.

Reply:
I've generally been leaving the min version off. There aren't too many old and broken versions laying around, like setuptools.

Reply:
This was autogenerated. Yes, I think the tests are failing for python 3.6 because of this. Thanks!